### PR TITLE
Add deploy script and optimism/base deployment

### PR DESCRIPTION
**Pull request by hexonaut**
Adding a deploy script as well as I deployed on Optimism+Base, set the max dsr to 1000% and burned the admin access.

Coverage is failing this should be addressed in a separate PR.

### CODE REVIEW

**Review by barrutko — DISMISSED** [review]

**Review by barrutko — DISMISSED** [review]
LGTM

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+contract Deploy is Script {
+
+    address internal constant MCD_POT = 0x197E90f9FAD81970bA7976f33CbD77088E5D7cf7;
+
+    uint256 internal constant DSR_1000APY = 1.000000076036763190083298291e27;
+
+    function deploy(string memory remoteRpcUrl) internal {
+        address deployer = msg.sender;
+        address admin = vm.envOr("ORACLE_ADMIN", address(0));
```
Comment (lucas-manuel) [inline script/Deploy.s.sol]: Align

File: script/Deploy.s.sol [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        vm.createSelectFork(remoteRpcUrl);
+
+        uint256 nonce = vm.getNonce(deployer);
+
+        vm.createSelectFork(getChain("mainnet").rpcUrl);
+
+        vm.startBroadcast();
+        address expectedReceiver = vm.computeCreateAddress(deployer, nonce + 1);
+        address forwarder = deployForwarder(expectedReceiver);
```
Comment (lucas-manuel) [inline script/Deploy.s.sol]: Align

File: script/Deploy.s.sol [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        if (admin != address(0)) {
+            oracle.grantRole(oracle.DEFAULT_ADMIN_ROLE(), admin);
+        }
+        oracle.renounceRole(oracle.DEFAULT_ADMIN_ROLE(), deployer);
+        vm.stopBroadcast();
+
+        console.log("Deployed Forwarder at:", forwarder);
+        console.log("Deployed Receiver at:", receiver);
+        console.log("Deployed DSRAuthOracle at:", address(oracle));
+        console.log("Deployed DSRBalancerRateProviderAdapter at:", address(adapter));
```
Comment (lucas-manuel) [inline script/Deploy.s.sol:62]:
```suggestion
        console.log("Deployed Forwarder at:                     ", forwarder);
        console.log("Deployed Receiver at:                      ", receiver);
        console.log("Deployed DSRAuthOracle at:                 ", address(oracle));
        console.log("Deployed DSRBalancerRateProviderAdapter at:", address(adapter));
```

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Small formatting changes

File: script/Deploy.s.sol (hunk 1 above)
Comment (hexonaut) [reply script/Deploy.s.sol]: done

File: script/Deploy.s.sol (hunk 2 above)
Comment (hexonaut) [reply script/Deploy.s.sol]: done

File: script/Deploy.s.sol (hunk 3 above)
Comment (hexonaut) [reply script/Deploy.s.sol:62]: done

**Review by barrutko — DISMISSED** [review]

File: script/Deploy.s.sol (hunk 3 above)
Comment (lucas-manuel) [reply script/Deploy.s.sol:62]: Suggestion adds spaces in the first param string so that the console output is aligned as well

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
One comment

File: script/Deploy.s.sol (hunk 3 above)
Comment (hexonaut) [reply script/Deploy.s.sol:62]: ah ok, done

**Review by barrutko — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**